plugin/src/main/kotlin/com/slapin/gsp/internal/api/SamsungApiClient.kt

```diff
@@ -141,7 +158,13 @@ internal class SamsungApiClient(
         .requestBuilder()
         .post(body)
         .build()
-    return newCall(request).execute().use { response -> response.body.fromJson() }
+    return newCall(request).execute().use { response ->
+      //submit-app don't have response body so only return boolean with successful status
+      when (T::class) {
+        Boolean::class -> response.isSuccessful as T
```

Comment:
I'd say it's better to extract `OkHttpClient.post(...): Response` method and use it in `submitApp` directly.